force-app/main/default/classes/GiftBatch_TEST.cls

```diff
@@ -190,12 +190,123 @@ private class GiftBatch_TEST {
         ElevateBatchServiceMock elevateBatchServiceMock = new ElevateBatchServiceMock();
         giftBatch.elevateBatchService = stubFor(elevateBatchServiceMock);
 
-        giftBatch.processGifts();
+        giftBatch.prepareGiftsForProcessing();
 
         System.assertEquals(true, elevateBatchServiceMock.enqueuePaymentCaptureJobForCalled);
     }
 
-    static String DUMMY_DATA_IMPORT_MODEL = '{"totalRowAmount":300.0,"totalCountOfRows":2}';
+    @IsTest
+    static void shouldUpdateProcessingGiftsInAbortedBatchesToFailed() {
+        GiftBatch giftBatch = new GiftBatch();
+        giftBatch.giftBatchId = DUMMY_GIFT_BATCH_ID;
+        giftBatch.gifts = stubFor(new GiftsMock());
+        giftBatch.gifts.dataImportsById = buildDataImportsByIds(DUMMY_GIFT_BATCH_ID.value());
+        giftBatch.gifts.giftsToUpdate = new List<DataImport__c>();
+        giftBatch.asyncApexJob =  new TestAsyncApexJob()
+            .withStatus(UTIL_BatchJobService.ApexJobStatus.ABORTED.name())
+            .build();
+
+        giftBatch.validate();
+
+        for(DataImport__c gift : giftBatch.gifts.giftsToUpdate) {
```

Comment:
We may be getting false-positives in the new unit tests. The giftBatch.gifts.giftsToUpdate list is coming back as empty to me. Could we update these unit tests to explicitly reference item(s) in the giftsToUpdate list when asserting?

Reply:
Updated!